features/step_definitions/password_steps.rb

```diff
@@ -56,3 +56,28 @@ def visit_url_in_email(email, subject)
   click_on "Send instructions"
   page.should have_selector(RECAPTCHA_SCRIPT, visible: false)
 end
+
+Then "it should not be possible to reset their password" do
+  assert_not has_link?('Forgot password?', visible: :all)
+
+  expect do
+    visit path_to('the provider reset password page') # FIXME: first time does not raise??
+    visit path_to('the provider reset password page')
+  end
+    .to raise_error(ActionController::RoutingError)
+end
+
+# TODO: do these programmatically
```

Comment:
What do you mean?

Reply:
Instead of submitting the form and sending the email and everything, generate the token directly if possible.

Reply:
I think it can be done like this:

```
  user.generate_lost_password_token!

  url = provider_password_url password_reset_token: user.lost_password_token, host: @domain
  visit url
```

But probably needs to be split in multiple steps:
1. Generate the token
2. Visit the URL with the token

Reply:
I think it's perfect, and `I follow the link found in the provider password reset email send to` can be used a a step.